tests/phpunit/tests/html-api/wpHtmlTagProcessor.php

```diff
@@ -476,6 +476,78 @@ public function test_get_updated_html_applies_updates_to_content_after_seeking_t
 		$this->assertSame( '<div wonky><img hidden></div>', $processor->get_updated_html() );
 	}
 
+	/**
+	 * Ensures that bookmarks start and langth correctly describe a given token in HTML.
+	 *
+	 * @ticket 61301
+	 *
+	 * @dataProvider data_html_nth_token_substring
+	 *
+	 * @param string $html            Input HTML.
+	 * @param string $match_nth_token The number of times to call ::match_token.
+	 * @param string $expected_match  The expected HTML at the matched token.
+	 */
+	public function test_token_bookmark_span( string $html, int $match_nth_token, string $expected_match ) {
+		$processor = new class( $html ) extends WP_HTML_Tag_Processor {
+			public function get_token_string() {
```

Comment:
not that it matters, but for what it's worth, I've been naming this `get_raw_token()` in multiple other places

Reply:
updated